sparkle/src/components/NewDropdown.tsx

```diff
@@ -0,0 +1,277 @@
+}
+
+type Simplify<T> = { [K in keyof T]: T[K] };
+
+type EitherChildrenOrProps<BaseProps, ExtraProps> =
+  | (BaseProps & ExtraProps & { children?: never })
+  | (BaseProps & { [K in keyof ExtraProps]?: never });
+
+type MutuallyExclusiveProps<BaseProps, ExtraProps> = Simplify<
+  EitherChildrenOrProps<BaseProps, ExtraProps>
+>;
+
+const NewDropdownMenuSubTrigger = React.forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.SubTrigger>,
+  MutuallyExclusiveProps<
+    React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubTrigger>,
+    LabelAndIconProps
+  > & {
+    inset?: boolean;
+  }
+>(({ className, label, icon, children, inset, ...props }, ref) => (
+  <DropdownMenuPrimitive.SubTrigger
+    ref={ref}
+    className={cn(
+      menuStyleClasses.item,
+      inset ? menuStyleClasses.inset : "",
+      className || ""
+    )}
+    {...props}
+  >
+    {label && (
+      <>
+        {icon && <Icon size="xs" visual={icon} />}
+        {label}
+        <span className={menuStyleClasses.subTrigger.default}>
+          <Icon size="xs" visual={ChevronRightIcon} />
+        </span>
+      </>
+    )}
+    {children}
+  </DropdownMenuPrimitive.SubTrigger>
+));
+NewDropdownMenuSubTrigger.displayName =
+  DropdownMenuPrimitive.SubTrigger.displayName;
+
+const NewDropdownMenuSubContent = React.forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.SubContent>,
+  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubContent>
+>(({ className, ...props }, ref) => (
+  <DropdownMenuPrimitive.SubContent
+    ref={ref}
+    className={cn(menuStyleClasses.container, "s-shadow-lg", className || "")}
+    {...props}
+  />
+));
+NewDropdownMenuSubContent.displayName =
+  DropdownMenuPrimitive.SubContent.displayName;
+
+const NewDropdownMenuContent = React.forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Content>
+>(({ className, sideOffset = 4, ...props }, ref) => (
+  <DropdownMenuPrimitive.Portal>
+    <DropdownMenuPrimitive.Content
+      ref={ref}
+      sideOffset={sideOffset}
+      className={cn(menuStyleClasses.container, "s-shadow-md", className || "")}
+      {...props}
+    />
+  </DropdownMenuPrimitive.Portal>
+));
+NewDropdownMenuContent.displayName = DropdownMenuPrimitive.Content.displayName;
+
+const NewDropdownMenuItem = React.forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.Item>,
+  MutuallyExclusiveProps<
+    React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Item> & {
+      inset?: boolean;
+    },
+    LabelAndIconProps & { description?: string }
+  >
+>(({ children, description, className, inset, icon, label, ...props }, ref) => (
+  <DropdownMenuPrimitive.Item
+    ref={ref}
+    className={cn(
+      menuStyleClasses.item,
+      inset ? menuStyleClasses.inset : "",
+      className || ""
+    )}
+    {...props}
+  >
+    {label && (
+      <div className="s-flex s-flex-row s-gap-1">
```

Comment:
Could we add `s-items-center`?

Reply:
Seen with Ed, we want to keep the icon aligned with the title.

Reply:
Updated, good catch!